src/components/DrawerContent/DrawerContent.tsx

```diff
@@ -11,14 +15,55 @@ const StyledDiv = styled.div`
 type DrawerContentProps = DrawerTopBarProps & DrawerBottomProps;
 
 function DrawerContent(props: DrawerContentProps) {
+
+	const { t } = useMosaicTranslation();
+
+	const [drawerButtons, setDrawerButtons] = useState<ButtonProps[]>([]);
+
+	useEffect(() => {

```

Comment:
This is a good approach but I have 2 recommendations here:
1. Change useEffect to useMemo (https://reactjs.org/docs/hooks-reference.html#usememo). This stores the result in memory and doesn't reassign it with every render, only when those props change. The way it is now it will re-render / re-create the buttons every single time any of it's parent components re-renders.
2. I don't like depending on the state because useState is asynchronous, this means the real value won't always be the rendered value because react has to wait for the component to finish re-rendering to update the state. Using useMemo also solves this issue because it assigns the value to an actual variable.

Reply:
Here's how the result would look like:
```suggestion
	const drawerButtons = useMemo(() => {
	        let buttons: ButtonProps[] = [];
	        ... all your code
	        return buttons;
	        }, [the deps you already have]);
```